packages/shared/src/contexts/ErrorReportingContext.tsx

```diff
@@ -0,0 +1,130 @@
+import { ScopeContext } from '@sentry/types';
+import { createContext, memo, ReactNode, useCallback, useContext, useRef } from 'react';
+import { graphql, useLazyLoadQuery } from 'react-relay';
+
+import { ErrorReportingContextQuery } from '~/generated/ErrorReportingContextQuery.graphql';
+
+export type AdditionalContext = Partial<Pick<ScopeContext, 'tags' | 'level' | 'user'>>;
+
+export type ReportFn = (
+  errorOrMessage: Error | string,
+  additionalContext?: AdditionalContext
+) => void;
+
+export const ErrorReportingContext = createContext<ReportFn | undefined>(undefined);
+
+/**
+ * USAGE GUIDELINES
+ *
+ * You can either send a javascript error object OR a string to sentry.
+ * It is *HIGHLY ENCOURAGED* to send the error itself as opposed to a string
+ * if possible, as it'll provide Sentry with a detailed stacktrace:
+ *
+ *    const reportError = useReportError();
+ *    try {
+ *      badFunction();
+ *    } catch (error) {
+ *      reportError(error); <-- pass along the caught error
+ *    }
+ *
+ * However, there may be cases where you don't have access to a literal
+ * error but still want to report it. In that case you can pass in a string:
+ *
+ *    const reportError = useReportError();
+ *    if (dataIsGood) {
+ *      // success case
+ *    } else {
+ *      reportError('data is messed up'); <-- send a string
+ *    }
+ *
+ * You can also provide an additional object param to provide more context.
+ * These options include:
+ *   - level: the severity of the error. these are enums defined on the Severity type in @sentry/types
+ *   - tags: optional metadata that might be useful
+ *
+ *      try {
+ *        badFunction();
+ *      } catch (error) {
+ *        reportError(error, {
+ *          level: 'fatal',
+ *          tags: {
+ *            context: 'onboarding',
+ *            step: 2
+ *          }
+ *        })
+ *      }
+ */
+export const useReportError = () => {
+  const reportError = useContext(ErrorReportingContext);
+
+  if (!reportError) {
+    throw new Error('Attempted to use ErrorReportingContext without a provider!');
+  }
+
+  return reportError;
+};
+
+type Props = {
+  setUser: (user: { id?: string }) => void;
+  captureMessage: (message: string, additionalContext?: AdditionalContext) => void;
+  captureException: (error: Error, additionalContext?: AdditionalContext) => void;
+  children: ReactNode;
+};
+
+const ErrorReportingProvider = memo(
+  ({ children, captureException, captureMessage, setUser }: Props) => {
+    console.log('Shared', useLazyLoadQuery);
+
+    const query = useLazyLoadQuery<ErrorReportingContextQuery>(
+      graphql`
+        query ErrorReportingContextQuery {
+          viewer {
+            ... on Viewer {
+              user {
+                dbid
+              }
+            }
+          }
+        }
+      `,
+      {}
+    );
```

Comment:
We might want to move this around so it doesn't block the initial load of the app